### PR TITLE
PyTorch 1.9 FastSAM `sum(0).sum()` predictor fix

### PR DESCRIPTION
**Pull request by Y-T-G**
Maybe fixes inplace error

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Improves FastSAM compatibility with older PyTorch versions by adding a targeted workaround for a mask summation bug in PyTorch 1.9, ensuring stable prompt-based segmentation across environments. ✅

### 📊 Key Changes
- Added a new version flag `TORCH_1_10` in `ultralytics/utils/torch_utils.py` for precise PyTorch version checks.
- Updated `FastSAM.predict.prompt` to conditionally compute mask sums:
  - Uses `masks[i].sum()` on PyTorch ≥ 1.10.
  - Falls back to `masks[i].sum(0).sum()` on PyTorch 1.9 to avoid a known bug.
- Minor import update to use the new `TORCH_1_10` flag.

### 🎯 Purpose & Impact
- Ensures FastSAM’s mask filtering works correctly on PyTorch 1.9, preventing incorrect removal of valid masks or potential errors. 🛠️
- Maintains identical behavior on modern PyTorch versions (≥ 1.10), so no changes for most users. 🚀
- Adds a reusable version check utility for future compatibility fixes. ♻️

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response — an Ultralytics engineer will review and assist soon. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Since this aims to address a potential in-place error in FastSAM, please add context on where/when it occurs.
- ✅ **Provide an MRE (if bug fix)**: If this PR fixes a bug, please include a Minimum Reproducible Example (MRE) that demonstrates the original issue and shows how this change resolves it. A short snippet or command is ideal, including device info (CPU/GPU/MPS) and versions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes (e.g., a FastSAM unit test validating mask-size filtering behavior and device coverage), and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

Additional notes for this PR:
- Since you’re switching to `torch.sum`, please ensure `import torch` is present in `ultralytics/models/fastsam/predict.py`. If it’s missing, kindly add it at the top of the file to avoid NameError on import.
- If the change affects device behavior (CPU/GPU/MPS), please note tested devices in the PR description and, if possible, include the MRE results across them. 🧪
- Consider adding a brief comment near the code explaining the rationale for `torch.sum` to aid future maintainers. ✨

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Comment by Y-T-G** [comment]
Doesn't work

**Comment by glenn-jocher** [comment]
@Y-T-G can you try https://github.com/ultralytics/ultralytics/pull/22170?

**Comment by glenn-jocher** [comment]
@Y-T-G got it, good fix. Simplified slightly.

**Comment by UltralyticsAssistant** [comment]
🎉 Merged—awesome work, @Y-T-G, with valuable input from @glenn-jocher! 

“Great things are done by a series of small things brought together.” — Vincent van Gogh. This precise workaround and the new `TORCH_1_10` flag in [PR #22169](https://github.com/ultralytics/ultralytics/pull/22169) exemplify that spirit—keeping FastSAM prompt-based segmentation stable on PyTorch 1.9 while maintaining identical behavior on newer versions. Thoughtful, user-centric engineering like this strengthens our ecosystem across diverse environments.

Quick nod to the clarity of the implementation:
```python
from ultralytics.utils.torch_utils import TORCH_1_10

mask_sum = masks[i].sum() if TORCH_1_10 else masks[i].sum(0).sum()
```

Huge thanks for the attention to detail and future-ready design. Your contributions elevate the experience for the whole community—keep them coming! 🚀
